torchdata/datapipes/map/util/converter.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+from typing import List, Optional
+
+from torch.utils.data import IterDataPipe, MapDataPipe
+
+
+# @functional_datapipe("to_iter_datapipe")  # This line must be kept for .pyi signature parser
+class MapToIterConverterIterDataPipe(IterDataPipe):
+    def __init__(self, datapipe: MapDataPipe, indices: Optional[List] = None):
```

Comment:
Any thoughts on requiring `indices` to be a `List` rather than say `Iterable`?

Taking in an `Iterable` complicates things when we have to serialize since in theory a generator can be passed. Then, we will have to materialize it, but that is only possible from we haven't started reading the generator already.